spirv_msl.cpp

```diff
@@ -4953,7 +4953,7 @@ void CompilerMSL::emit_custom_functions()
 			{
 				uint32_t dimensions = spv_func - SPVFuncImplArrayCopyMultidimBase;
 				string tmp = "template<typename T";
-				for (uint8_t i = 0; i < dimensions; i++)
+				for (uint32_t i = 0; i < dimensions; i++)
 				{
 					tmp += ", uint ";
 					tmp += 'A' + i;
```

Comment:
Should probably explicitly cast this to `char('A' + i)`. Implicit type promotion here seems dicey.

Reply:
Yes, and it may be need a `assert(i <= 0xff);`